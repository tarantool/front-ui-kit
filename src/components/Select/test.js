import React from 'react';
import renderer from 'react-test-renderer';

import { Select } from './index';

const noop = () => void 0;

describe('Select', () => {
  it('Render empty', () => {
    const tree = renderer
      .create(
        <>
          <Select options={[]} value={undefined} onChange={noop} />
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Render with data', () => {
    const tree = renderer
      .create(
        <>
          <Select options={[{ label: 'Male', value: 'male' }]} value={'male'} onChange={noop} />
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Render with data and allow search', () => {
    const tree = renderer
      .create(
        <>
          <Select options={[{ label: 'Male', value: 'male' }]} value={'male'} onChange={noop} allowSearch />
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
