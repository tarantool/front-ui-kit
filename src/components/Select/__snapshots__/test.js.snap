// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Render empty 1`] = `
<div
  className="css-1gomreu"
  onClick={[Function]}
>
  <div
    className="css-1ondgiw"
  >
    <input
      autocomplete="new-password"
      className="css-16fuuzx"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder=""
    />
    <div
      className="css-qidtvj"
    >
      <svg
        className="css-1ews5mj"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Select Render with data 1`] = `
<div
  className="css-1gomreu"
  onClick={[Function]}
>
  <div
    className="css-1ondgiw"
  >
    <input
      autocomplete="new-password"
      className="css-16fuuzx"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder=""
      value="Male"
    />
    <div
      className="css-qidtvj"
    >
      <svg
        className="css-1ews5mj"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Select Render with data and allow search 1`] = `
<div
  className="css-1gomreu"
  onClick={[Function]}
>
  <div
    className="css-8o7sw6"
  >
    <input
      autocomplete="new-password"
      className="css-16fuuzx"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder=""
      value="Male"
    />
    <div
      className="css-qidtvj"
    >
      <svg
        className="css-1ews5mj"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </div>
  </div>
</div>
`;
